plutus-ledger-api/exe/analyse-script-events/README.md

```diff
@@ -0,0 +1,134 @@
+## Analysis of mainnet script events
+
+The `analyse-script-events` executable provides some simple analyses of on-chain
+script events.  In order to use this you must have a dump of the script events
+that have occurred on the Cardano blockchain: instructions on how to do this can
+be found in
+[test-onchain-evaluation/README.md](../test-onchain-evaluation/README.md).  The
+dump consists of a number of `.bz2` files with names like
+
+```
+0000000105908766-ae0fda07afb26a9c0636e7578cc10e379393ed830e7f7574d60032004da6287a.event.bz2
+```
+
+The first part of the name specifies the slot containing the first script event
+in the file; the number of events per file is not fixed, but the more recent
+files typically contain about 50,000 events.  The files contain scripts which
+have been run on the chain, together with their inputs encoded using the
+[Data](https://github.com/input-output-hk/plutus/blob/master/plutus-core/plutus-core/src/PlutusCore/Data.hs)
+type. For standard validation scripts the inputs consist of datum, redeemer, and
+script context; for minting policies the inputs are a redeemer and script
+context.
+
+To run an analysis type
+```
+cabal run analyse-script-events <dir> <analysis>.
+```
+
+The `dir` argument specifies the path to a directory containing uncompressed
+`.event` files (possibly the full set, or perhaps just a subset).  It may take
+several hours for the analysis to complete depending on the complexity of the
+analysis and the number of files to be analysed.
```

Comment:
The full dump takes up 136G, so I think it may be difficult to speed it up.  What really takes a long time is when you're doing something that requires you to deserialise all of the scripts and traverse the ASTs.

Reply:
> Would it be faster if we did the analysis in R and dumped the data into some intermediate format, like a massive CSV file?

I don't think that makes sense.  The basic data is a big bunch of containing transaction info, and you have to process them using Haskell to get hold of the stuff you're interested in before you even start thinking about R or something similar (I just did some of the initial analysis using `grep/sed/awk/sort` etc.).